core/eolearn/core/constants.py

```diff
@@ -292,7 +293,30 @@ class FeatureTypeSet(metaclass=DeprecatedCollectionClass):
     RASTER_TYPES_1D = frozenset([FeatureType.SCALAR_TIMELESS, FeatureType.LABEL_TIMELESS])
 
 
-class OverwritePermission(Enum):
+def _warn_and_adjust_permissions(name: T) -> T:
+    if isinstance(name, str) and name.upper() == "OVERWRITE_PATCH":
+        warnings.warn(
+            '"OVERWRITE_PATCH" permission is deprecated and will be removed in a future version',
+            category=EODeprecationWarning,
+            stacklevel=3,
+        )
+    return name
+
+
+class PermissionsWithDeprecations(EnumMeta):
+    """A custom EnumMeta class for catching the deprecated Enum members of the FeatureType Enum class."""
+
+    def __getattribute__(cls, name: str) -> Any:  # noqa[N805]
+        return super().__getattribute__(_warn_and_adjust_permissions(name))
+
+    def __getitem__(cls, name: str) -> Any:  # noqa[N805]
+        return super().__getitem__(_warn_and_adjust_permissions(name))
+
+    def __call__(cls, value: str, *args: Any, **kwargs: Any) -> Any:  # noqa[N805]
+        return super().__call__(_warn_and_adjust_permissions(value), *args, **kwargs)
```

Comment:
perhaps it would make sense to change the `name` parameter to `permission` or something. `name` is still used from the FeatureType enum

Reply:
those are taken from the Enum implementation. The `name` is the name of the enum constant. I dont think we should be losing sleep over this one